### PR TITLE
Use plain string certificates

### PR DESCRIPTION
**Pull request by nelsonkopliku**
# Description

This PR enables using SUMA ca certificate as a plain string instead of a reference to a file in the filesystem.

Took inspiration from this post https://elixirforum.com/t/using-client-certificates-from-a-string-with-httposion/8631/7

## How was this tested?

Current tests updated + manual tests.

### CODE REVIEW

**Review by jamie-suse — APPROVED** [review]
wow, so much nicer :rocket:

File: lib/trento/infrastructure/software_updates/adapter/suma_http_executor.ex [hunk 1]
```diff
@@ -3,44 +3,42 @@ defmodule Trento.Infrastructure.SoftwareUpdates.Suma.HttpExecutor do
...
-              use_ca_cert :: boolean()
+              ca_cert :: String.t() | nil
             ) ::
               {:ok, HTTPoison.Response.t()} | {:error, HTTPoison.Error.t()}
 
   @behaviour Trento.Infrastructure.SoftwareUpdates.Suma.HttpExecutor
 
   @impl true
-  def login(base_url, username, password, use_ca_cert \\ false) do
+  def login(base_url, username, password, ca_cert \\ nil) do
```
Comment (dottorblaster) [inline lib/trento/infrastructure/software_updates/adapter/suma_http_executor.ex:41]: Do we need the `\\ nil`? I mean, we can just have `ca_cert`, it doesn't really convert to some default if it's `nil` :+1:

**Review by dottorblaster — CHANGES_REQUESTED** [review]
Just a comment about default arguments but I think this is ready to ship

**Review by dottorblaster — APPROVED** [review]
LGTM

**Pull request merged**